src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -47,11 +47,6 @@ internal SyntaxTreeSemanticModel(CSharpCompilation compilation, SyntaxTree synta
             _syntaxTree = syntaxTree;
             _ignoresAccessibility = ignoreAccessibility;
 
-            if (!this.Compilation.SyntaxTrees.Contains(syntaxTree))
-            {
-                throw new ArgumentOutOfRangeException(nameof(syntaxTree), CSharpResources.TreeNotPartOfCompilation);
-            }
```

Comment:
nit: consider adding a debug assert instead.

Reply:
You commented at the same time I mentioned why I didn't :wink:

https://github.com/dotnet/roslyn/pull/62382#discussion_r914184981